### PR TITLE
Chore: Update `unleash-proxy-client`

### PR DESCRIPTION
**Pull request by tuliomir**
This upgrade removes some obsolete indirect dependencies that were interfering with an upgrade to Node.js `v20`

### Acceptance Criteria
- Upgrades [`unleash-proxy-client`](https://github.com/Unleash/unleash-proxy-client-js) to the latest stable version `3.2.0`

### Breaking changes
On `v2.0.2` [release](https://github.com/Unleash/unleash-proxy-client-js/releases/tag/v2.0.2-beta-0) it was announced that this lib will no longer support natively `react native async storage`. This native support was precisely what was breaking our migration, so it is a welcome change for this repository.

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -3375,14 +3375,6 @@
...
       }
     },
-    "@react-native-async-storage/async-storage": {
-      "version": "1.17.6",
-      "resolved": "https://registry.npmjs.org/@react-native-async-storage/async-storage/-/async-storage-1.17.6.tgz",
-      "integrity": "sha512-XXnoheQI3vQTQmjphdXNLTmtiKZeRqvI8kPQ25X5Eae7nZjdYEEGN+0z8N2qyelbUIQwKgmW0aagJk56q7DyNg==",
-      "requires": {
-        "merge-options": "^3.0.4"
-      }
-    },
```
Comment (r4mmer) [inline package-lock.json:3385]: Can you check why was this removed?
Better yet, why did we have this dependency since this is not a react native environment

Comment (tuliomir) [reply package-lock.json:3385]: This was removed precisely because it was not needed. See https://github.com/Unleash/unleash-proxy-client-js/pull/76

The storage now defaults to `window.localStorage`

**Review by r4mmer — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]
